src/embeddings/embeddings_calculator.cc

```diff
@@ -123,42 +128,76 @@ class EmbeddingsCalculator : public CalculatorBase {
                     ov::Shape{received_batch_size},
                     strings->data()};
                 tokenizerOutputMap = tokenizer_session->infer(tokenizerInputMap);
-            } else {
-                // TODO: input already tokenized
-                return absl::InvalidArgumentError("not implemented");
-            }
-            ::InferenceInput embeddingsInputMap;
-            // Check if tokenizer produced at least the number of outputs as there are inputs in embedding model
-            RET_CHECK(tokenizerOutputMap.size() >= embeddingsInputNames.size());
-            for (const auto& embeddingsInputName : embeddingsInputNames) {
-                auto it = tokenizerOutputMap.find(embeddingsInputName);
-                RET_CHECK(it != tokenizerOutputMap.end());
-                SPDLOG_LOGGER_DEBUG(embeddings_calculator_logger, "Embedding model input {} is connected with matching tokenizer output", embeddingsInputName);
-                embeddingsInputMap[embeddingsInputName] = it->second;
-                if (embeddingsInputName == "attention_mask") {
-                    if (received_batch_size == 1) {
-                        handler.setPromptTokensUsage(it->second.get_size());
-                        continue;
-                    }
-                    size_t attendedTokens = 0;
-                    if (it->second.get_element_type() == ov::element::Type_t::i64) {
-                        for (int i = 0; i < it->second.get_size(); i++) {
-                            attendedTokens += reinterpret_cast<int64_t*>(it->second.data())[i];
-                        }
-                    } else if (it->second.get_element_type() == ov::element::Type_t::i32) {
-                        for (int i = 0; i < it->second.get_size(); i++) {
-                            attendedTokens += reinterpret_cast<int32_t*>(it->second.data())[i];
+                // Check if tokenizer produced at least the number of outputs as there are inputs in embedding model
+                RET_CHECK(tokenizerOutputMap.size() >= embeddingsInputNames.size());
+                RET_CHECK(tokenizerOutputMap[EMBEDDINGS_MODEL_INPUT_IDS_NAME].get_shape().size() == 2);
```

Comment:
Again, not sure where my previous comment go, but this line may insert element with key `EMBEDDINGS_MODEL_INPUT_IDS_NAME` to the tokenizerOutputMap map. This is what `operator[]` is doing when key does not exist

Reply:
If the output was not created then RET_CHECK one line earlier would fail. If the name of the input will not match then this RET_CHECK would fail anyway, right?

Reply:
Yes, but it is not the only reason for EMBEDDINGS_MODEL_INPUT_IDS_NAME to be absent. The inference could pass (making`tokenizerOutputMap.size() >= embeddingsInputNames.size()` statement true), but still EMBEDDINGS_MODEL_INPUT_IDS_NAME might not be there